### ISSUE TITLE
Password can be the e-mail

### ISSUE THREAD

**Issue by louis123562** [PHPAuth/PHPAuth#421]
Hey,

i figured out that the password is not compared with the e-mail address of a user when using function:
`addUser` or `resetPass` or `updateUser`.

So it is possible to have the followding credentials:

```
mail: mail@example.org
pass: mail@example.org
```

What are you thoughts? I guess that we should prevent using the same email as the password for a user-account.

**Comment by KarelWintersky** [comment]
Why not? It is not a vulnerability.

**Comment by louis123562** [comment]
I think one of the methods a "hacker" would always try is to use the mail as password, because users find it very comfortable, if it is possible to use email as password. But this is only my opinion.

**Comment by turbopixel** [comment]
When I start developing, I always have an account where the email equals the password for testing purposes. 
But in production mode, nobody should do it!

**Comment by KarelWintersky** [comment]
> I think one of the methods a "hacker" would always try is to use the mail as password, because users find it very comfortable,

It's not an our problem. 
Also, we can add this as option to configuration (with default value: false). 

Next step: use validation plugins for this - as composer packages. Anybody knows how to do this? I am not :(

Something like
```
PHPAuth::init( pdo: $pdo_connection, config: $config_instance, validators: [ PHPAuthValidator\PasswordEqEmail::class, PHPAuthValidator\PasswordFromDictionary::class ] );
```

**Comment by Conver** [comment]
We already use zxcvbn to validate password, should be possible to extend it to validate that it's not the same email.

**Comment by louis123562** [comment]
What we do with zxcvbn is to make sure, that the password has a type X of strength for the password on its own; but what we should do is to make sure, that the password is not either contains any of the information in the row of the current user in phpauth_users table. That would be much more secure :)

**Comment by Conver** [comment]
Accepting PR :+1:

**Comment by Criptos** [comment]
Fixed for v2.0
